src/daft-scan/src/storage_config.rs

```diff
@@ -23,6 +23,16 @@ pub enum StorageConfig {
 }
 
 impl StorageConfig {
+    /// TODO: Arc the inner IOConfig on each StorageConfig instead to avoid needing to Arc here
+    pub fn get_io_config(&self) -> Arc<IOConfig> {
```

Comment:
I couldn't make it work because we have a bunch of code that requires `Arc<IOConfig>` (specifically I think `get_io_client` uses that), so if I return a reference I end up needing to clone/arc it somewhere upstream anyways...